### ISSUE TITLE
[object-curly-spacing] Rule does not apply to MappedType

### ISSUE THREAD

**Issue by MatthiasKunnen** [typescript-eslint/typescript-eslint#3175]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/object-curly-spacing": ["error", "never"]
  }
}
```

```TS
var d: { [k in 'string' | 'data']: any };
```

**Expected Result**
I am expecting the linter to complain about the space after the `{` and the space before the `}`.

**Actual Result**

The linter happily accepts this code.

**Additional Info**

I've tested this issue on 946ee3b68823a48fad2cb7b99589e56a68fb11e3 and could reproduce it.  
I'm working on a PR to first test, and then fix this.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.17.0` |
| `@typescript-eslint/parser`        | `4.17.0` |
| `TypeScript`                       | `4.2.2` |
| `ESLint`                           | `7.20.0` |
| `node`                             | `12.18.0` |

**Comment by bradzacher** [comment]
happy to accept a PR to add support!
